crates/neon/src/result/mod.rs

```diff
@@ -52,7 +52,21 @@ use crate::{context::Context, handle::Handle, types::Value};
 pub struct Throw(PhantomData<*mut ()>); // *mut is !Send + !Sync, making it harder to accidentally store
 
 impl Throw {
-    pub(crate) fn new() -> Self {
+    #[cfg(feature = "sys")]
+    /// Creates a `Throw` struct representing a JavaScript exception
+    /// state.
+    ///
+    /// # Safety
+    ///
+    /// `Throw` should *only* be constructed when the JavaScript VM is in a
```

Comment:
I really like how the safety requirements can be more concretely explained now in terms of the `sys` crate.